src/Illuminate/Http/Middleware/TrustHosts.php

```diff
@@ -53,6 +62,24 @@ public function handle(Request $request, $next)
         return $next($request);
     }
 
+    /**
+     * Specify the hosts that should be always trusted.
+     *
+     * @param  array<int, string>  $hosts
+     * @param  bool  $subdomains
+     * @return void
+     */
+    public static function at(array $hosts, bool $subdomains = true)
+    {
+        if ($subdomains) {
+            if ($host = parse_url(config('app.url'), PHP_URL_HOST)) {
+                $hosts[] = '^(.+\.)?'.preg_quote($host).'$';
+            }
```

Comment:
```suggestion
            $hosts[] = $this->allSubdomainsOfApplicationUrl();
```

Reply:
But can't be called statically. And change it, would be a breaking change for existing apps.